### PR TITLE
feat: #5367 app pipeline page

### PR DESCRIPTION
**Pull request by bashleigh**
# Pull request checklist

**Detail as per issue below (required):**

fixes: #5367 #5368 #5508

### CODE REVIEW

**Review by joshbalfour — APPROVED** [review]

File: packages/developer-portal/src/components/pages/app-detail/app-content-v8.tsx [hunk 1]
```diff
@@ -0,0 +1,217 @@
+import * as React from 'react'
```
Comment (willmcvay) [inline packages/developer-portal/src/components/pages/app-detail/app-content-v8.tsx:1]: Can you put all the v8 app detail stuff in a v8 folder please? Just to make it easier to work out what's what when we want to switch over from the old pages to the new ones. Ta

File: packages/developer-portal/src/components/pages/app-detail/app-content-v8.tsx [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+  }
+
+const AppContent: React.FC<AppContentProps> = ({ appDetailState }) => {
+  const appDetailData = appDetailState.data || {}
+
+  return (
+    <>
+      <Grid>
+        <ColSplit>
+          <InputWrap>
```
Comment (willmcvay) [inline packages/developer-portal/src/components/pages/app-detail/app-content-v8.tsx:113]: So I don't think you need input wrap - that's really for, well, inputs!

File: packages/developer-portal/src/components/pages/app-detail/app-content-v8.tsx [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+  const url = routes.APP_PREVIEW.replace(':appId', appId)
+  localStorage.setItem('developer-preview-app', JSON.stringify(appDetailState))
+  window.open(url, '_blank')
+}
+
+export const CustomUninstallCell: React.FC<{ onClick: () => void }> = ({ onClick }) => (
+  <a onClick={onClick}>Uninstall</a>
+)
+
+export const generateInstallationTableColumns =
```
Comment (willmcvay) [inline packages/developer-portal/src/components/pages/app-detail/app-content-v8.tsx:51]: All of these legacy functions can go right?

File: packages/developer-portal/src/components/pages/app-detail/app-detail-v8.tsx [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+                alt={appDetailData.name}
+              />
+            }
+            <Title>{appDetailData.name}</Title>
+          </FlexContainer>
+          <Tabs
+            isFullWidth
+            name="app_tabs"
+            onChange={(event) =>
+              // @ts-ignore
```
Comment (willmcvay) [inline packages/developer-portal/src/components/pages/app-detail/app-detail-v8.tsx:140]: Why ts-ignore? Can you not do `event.target.value ?? 'DEFAULT_TAB` or similar?

File: packages/developer-portal/src/components/pages/app-detail/pipeline/create-pipeline.tsx [hunk 5]
```diff
@@ -0,0 +1,229 @@
...
+    formState: { errors },
+  } = useForm<PipelineModelInterface>({
+    resolver: yupResolver(schema),
+    defaultValues: {
+      buildCommand: 'build',
+      packageManager: PackageManagerEnum.YARN,
+    },
+  })
+
+  const [loading, , send, submissionErrors] = useReapitUpdate<Partial<PipelineModelInterface>, PipelineModelInterface>({
```
Comment (willmcvay) [inline packages/developer-portal/src/components/pages/app-detail/pipeline/create-pipeline.tsx:84]: why the 2 commas? Are you not defining a var here?

File: packages/developer-portal/src/components/pages/app-detail/pipeline/create-pipeline.tsx [hunk 6]
```diff
@@ -0,0 +1,229 @@
...
+    const result = await createPipeline({
+      ...values,
+      appType: AppTypeEnum.REACT,
+      name: values.repository,
+    })
+
+    if (result) refresh()
+  }
+
+const PipelineCreationModal = ({
```
Comment (willmcvay) [inline packages/developer-portal/src/components/pages/app-detail/pipeline/create-pipeline.tsx:51]: I would probably move to it's own component

File: packages/developer-portal/src/components/pages/app-detail/pipeline/create-pipeline.tsx [hunk 7]
```diff
@@ -0,0 +1,229 @@
...
+import { WebhooksAnimatedDocsIcon } from '../../webhooks/webhooks-animated-docs-icon'
+import { ExternalPages, openNewPage } from '@/utils/navigation'
+import { mixed, object, string } from 'yup'
+import { AppTypeEnum, PackageManagerEnum, PipelineModelInterface } from '@reapit/foundations-ts-definitions'
+import { httpsUrlRegex, UpdateActionNames } from '@reapit/utils-common'
+import { useForm } from 'react-hook-form'
+import { yupResolver } from '@hookform/resolvers/yup'
+import errorMessages from '@/constants/error-messages'
+import { useReapitUpdate } from '@reapit/utils-react'
+import { reapitConnectBrowserSession } from '../../../../core/connect-session'
```
Comment (willmcvay) [inline packages/developer-portal/src/components/pages/app-detail/pipeline/create-pipeline.tsx:36]: Should come from @reapit/connect-session

File: packages/developer-portal/src/components/pages/app-detail/pipeline/deployment-info.tsx [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+  }
+}
+
+const buildStatusToReadable = (status: string): string =>
+  status
+    .split('_')
+    .map((str) => str.toLowerCase())
+    .join(' ')
+
+const PipelineInfo = ({ pipeline }: { pipeline: PipelineModelInterface }) => (
```
Comment (willmcvay) [inline packages/developer-portal/src/components/pages/app-detail/pipeline/deployment-info.tsx:44]: Can you define an interface for this please

File: packages/developer-portal/src/components/pages/app-detail/pipeline/create-pipeline.tsx (hunk 6 above)
Comment (willmcvay) [reply packages/developer-portal/src/components/pages/app-detail/pipeline/create-pipeline.tsx:51]: And defining an interface for the props is a bit more readable

File: packages/developer-portal/src/constants/action-types.ts [hunk 9]
```diff
@@ -10,6 +10,16 @@ const ActionTypes = {
   FETCH_APP_LIST_FAILED: 'FETCH_APP_LIST_FAILED',
   CLEAR_APP_LIST: 'CLEAR_APP_LIST',
 
+  FETCH_APP_PIPELINE: 'FETCH_APP_PIPELINE',
```
Comment (willmcvay) [inline packages/developer-portal/src/constants/action-types.ts:13]: Do you need these?

File: packages/utils-common/src/reapit-data/get-fetcher.ts [hunk 10]
```diff
@@ -16,13 +17,19 @@ export const getFetcher = async <DataType>({
   action,
   connectSession,
   queryParams,
+  uriParams,
   headers,
   logger,
 }: GetFetcherParams): Promise<DataType | string> => {
   const getAction = getActions[action]
   const { api, path } = getAction
+  const deSerialisedPath = uriParams
```
Comment (willmcvay) [inline packages/utils-common/src/reapit-data/get-fetcher.ts:26]: Probably worth breaking out and testing

File: packages/utils-react/src/use-reapit-data/use-reapit-update.ts [hunk 11]
```diff
@@ -0,0 +1,101 @@
...
+  const [error, setError] = useState<string | null>(null)
+  const { connectSession } = useReapitConnect(reapitConnectBrowserSession)
+  const [data, setData] = useState()
+  const { error: errorSnack } = useSnack()
+  const [success, setSuccess] = useState<undefined | boolean>(undefined)
+
+  const send: SendFunction<ParamsType> = async (params: ParamsType): Promise<boolean> => {
+    const updateAction = updateActions[action]
+    const { api, path } = updateAction
+    const deSerialisedPath = uriParams
```
Comment (willmcvay) [inline packages/utils-react/src/use-reapit-data/use-reapit-update.ts:49]: Same point as above, would probably be worth breaking out and testing

File: packages/developer-portal/src/components/pages/app-detail/app-content-v8.tsx (hunk 1 above)
Comment (willmcvay) [reply packages/developer-portal/src/components/pages/app-detail/app-content-v8.tsx:1]: File missing tests

File: packages/developer-portal/src/components/pages/app-detail/app-detail-v8.tsx [hunk 12]
```diff
@@ -0,0 +1,169 @@
+import React, { useState } from 'react'
```
Comment (willmcvay) [inline packages/developer-portal/src/components/pages/app-detail/app-detail-v8.tsx:1]: File missing tests

File: packages/developer-portal/src/components/pages/app-detail/pipeline/create-pipeline.tsx [hunk 13]
```diff
@@ -0,0 +1,229 @@
+import React, { useState } from 'react'
```
Comment (willmcvay) [inline packages/developer-portal/src/components/pages/app-detail/pipeline/create-pipeline.tsx:1]: File missing tests

File: packages/developer-portal/src/components/pages/app-detail/app-content-v8.tsx (hunk 2 above)
Comment (willmcvay) [reply packages/developer-portal/src/components/pages/app-detail/app-content-v8.tsx:113]: I'm not too worried TBH, it's a placeholder file I appreciate

**Review by willmcvay — CHANGES_REQUESTED** [review]
Mostly small bits and tests, loads of good stuff in here though - useReapitUpdate looks 👍🏻

File: packages/developer-portal/src/components/pages/app-detail/pipeline/create-pipeline.tsx (hunk 5 above)
Comment (bashleigh) [reply packages/developer-portal/src/components/pages/app-detail/pipeline/create-pipeline.tsx:84]: yea, eslint will hate me for not using it so didn't declare it 😂

File: packages/developer-portal/src/components/pages/app-detail/pipeline/create-pipeline.tsx (hunk 6 above)
Comment (bashleigh) [reply packages/developer-portal/src/components/pages/app-detail/pipeline/create-pipeline.tsx:51]: (y)

File: packages/developer-portal/src/constants/action-types.ts (hunk 9 above)
Comment (bashleigh) [reply packages/developer-portal/src/constants/action-types.ts:13]: not any longer 👍

File: packages/developer-portal/src/components/pages/app-detail/pipeline/create-pipeline.tsx (hunk 7 above)
Comment (bashleigh) [reply packages/developer-portal/src/components/pages/app-detail/pipeline/create-pipeline.tsx]: that's `ReapitConnectBrowserSession` which is the class that's being created from core/connect-session

File: packages/developer-portal/src/components/pages/app-detail/pipeline/deployment-info.tsx (hunk 8 above)
Comment (bashleigh) [reply packages/developer-portal/src/components/pages/app-detail/pipeline/deployment-info.tsx]: does it need one?

File: packages/developer-portal/src/components/pages/app-detail/app-detail-v8.tsx (hunk 4 above)
Comment (bashleigh) [reply packages/developer-portal/src/components/pages/app-detail/app-detail-v8.tsx]: Can't seem to solve this. onChange gives a strange event where Target is of type 'T' and this generic type doesn't have value as a property

File: packages/developer-portal/src/components/pages/app-detail/app-content-v8.tsx (hunk 2 above)
Comment (bashleigh) [reply packages/developer-portal/src/components/pages/app-detail/app-content-v8.tsx]: Is there anything similar to this I can use instead?

**Review by willmcvay — APPROVED** [review]
LGTM 👍🏻

**Pull request merged**